### PR TITLE
第7、8、9周作业

### PR DESCRIPTION
**Pull request by wen2024**
麻烦老师简单看一下。

### CODE REVIEW

**Comment by wen2024** [comment]
老师能看下第10周作业里面的loader.py 下的 load_coupus_vocab 类函数吗，我是想实现把所有在语料中出现的字符和其在bert的词汇表做一个映射，得到一个新的词汇表，键是字符，值是在bert的词汇表中的label，但是这样做了之后，模型在最后计算loss那一步报错了，我怀疑是因为我的新词汇表只有3000多字符，所有最后其进行反向传播更新bert的embedded层权重时，2边的维度不一致导致的报错，因为bert的原本词汇表有2w多。请问老师是这样的吗？

**Comment by wen2024** [comment]
老师， mask应该是和x同维度的吗？我看老师的实现里面是有3个维度，和x 二维的并不相同。
mask = torch.tril(torch.ones((x.shape[0], x.shape[1])))
mask = mask.to(x.device)
x_encoder, _ = self.encoder(x, attention_mask=mask)
